examples/shader/post_process_pass.rs

```diff
@@ -0,0 +1,383 @@
+//! This example shows how to create a custom render pass that runs after the main pass
+//! and reads the texture generated by the main pass.
+//!
+//! The example shader is a very simple implementation of chromatic aberration.
+//!
+//! This is a fairly low level example and assumes some familiarity with rendering concepts and wgpu.
+
+use bevy::{
+    core_pipeline::{
+        clear_color::ClearColorConfig, core_3d,
+        fullscreen_vertex_shader::fullscreen_shader_vertex_state,
+    },
+    prelude::*,
+    render::{
+        extract_component::{
+            ComponentUniforms, ExtractComponent, ExtractComponentPlugin, UniformComponentPlugin,
+        },
+        render_graph::{Node, NodeRunError, RenderGraph, RenderGraphContext, SlotInfo, SlotType},
+        render_resource::{
+            BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+            BindGroupLayoutEntry, BindingResource, BindingType, CachedRenderPipelineId,
+            ColorTargetState, ColorWrites, FragmentState, MultisampleState, Operations,
+            PipelineCache, PrimitiveState, RenderPassColorAttachment, RenderPassDescriptor,
+            RenderPipelineDescriptor, Sampler, SamplerBindingType, SamplerDescriptor, ShaderStages,
+            ShaderType, TextureFormat, TextureSampleType, TextureViewDimension,
+        },
+        renderer::{RenderContext, RenderDevice},
+        texture::BevyDefault,
+        view::{ExtractedView, ViewTarget},
+        RenderApp,
+    },
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(AssetPlugin {
+            // Hot reloading the shader works correctly
+            watch_for_changes: true,
+            ..default()
+        }))
+        .add_plugin(PostProcessPlugin)
+        .add_startup_system(setup)
+        .add_system(rotate)
+        .add_system(update_settings)
+        .run();
+}
+
+/// It's generally encouraged to setup post processing effect as a plugin
+struct PostProcessPlugin;
+impl Plugin for PostProcessPlugin {
+    fn build(&self, app: &mut App) {
+        app
+            // The settings will be a component that lives in the main world but will be extracted to the render world every frame.
+            // This makes it possible to control the effect from the main world.
+            // This plugin will take care of extracting it automatically.
+            .add_plugin(ExtractComponentPlugin::<PostProcessSettings>::default())
+            // The settings will use a uniform buffer, so we need this plugin to let bevy manage this automatically.
+            .add_plugin(UniformComponentPlugin::<PostProcessSettings>::default());
+
+        // We need to get the render app from the main app
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        // The renderer has multiple stages.
+        // For more details of each stage see the docs for `RenderSet`
```

Comment:
I'd like to give a quick overview of each stage. As far as I understand it,
* Extract - copy components from the main to render world. Rendering is done in parallel in a separate ECS world, so we have to sync data from the main to render world in extract. Entities/components don't live past 1 frame in the render world, but resources do.
* Prepare - create resources like textures/specialized pipelines/uniforms/etc
* Queue - create bind groups using prepared resources, and also the phase item thing I'm not super familiar with
* Some others I've never interacted with

Really it would be better if we updated the docs/module docstrings to have all this info...

Reply:
The RenderSet docs does explain some of that which is why I mentioned it. I ended up removing that comment though, because there's nothing using the render stages bellow that comment.